### DISCUSSION TITLE
Make REPL type rendering less noisy; suppress type paths to imported names

### DISCUSSION THREAD

**Discussion by bjornregnell** [scala/scala3#17877] [Feature Requests]
When echoing values, the REPL shows all those long package names p1.p2.p3 (except for some scala. stuff like Vector) in front of all types even if I have imported p1.p2.p3 and the type is unique in the scope. 

It would be very nice if the REPL could be made to be less noisy and suppress package names etc. before types if they are imported or unambiguous.

Currently:
```
scala> import org.com.herandthere.mylongpackaname.MyType

scala> Vector(MyType(42))
val res0: Vector[org.com.herandthere.mylongpackaname.MyType] = Vector(MyType(42))
```

Could be as nice as:
```
scala> import org.com.herandthere.mylongpackaname.MyType

scala> Vector(MyType(42))
val res0: Vector[MyType] = Vector(MyType(42))
```

**Comment by bjornregnell** [comment]
This could be done for the REPL printing specifically, leaving out error messages, IDE type hoovering etc.

**Comment by som-snytt** [comment]
https://github.com/scala/bug/issues/8349

**Comment by bjornregnell** [comment]
Aha - thanks for linking to that issue, @som-snytt ! This would be a great increase in REPL-ergonomy also in Scala 2 :+1: DRY shells for the people!!

**Comment by erikerlandson** [comment]
This would be great for the REPL and even compiler errors
